### PR TITLE
Fix #1522: Context menu interfering with text selection menu.

### LINKED ISSUE #1522: Text highlight context menu missing
### Description: 
Long press on text isn't highlighting the text and bringing up the context menu. Follow up to #1512

### Steps to Reproduce 
  1. Clean install 1.12(19.09.10.13)
  2. Open any page
  3. Long press on any text on the page
  4. No text gets highlighted and no context menu is shown

**Actual result:** 
Text highlight context menu missing

**Expected result:**
Long press on text should highlight the text and show context menu

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.12(19.09.10.13)

**Device details:** 
iPhone XR (iOS 13.1)

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information
Follow up to #1512 where 2 context menu were showing.

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes issue #1522 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/internal/issues/new?assignees=&labels=security%2C+security-reviews&template=security-review.md&title=iOS:%20Security+Review+for) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/ContextMenu.js [hunk 1]
```diff
@@ -13,20 +13,25 @@ window.__firefox__.includeOnce("ContextMenu", function() {
...
+      // should not show the menu(only text selection menu if applicable).
+      webkit.messageHandlers.contextMenuMessageHandler.postMessage(data);
       return;
     }
 
-    var data = {};
-
     if (targetLink) {
       data.link = targetLink.href;
+      data.title = targetLink.textContent;
```
Comment (iccub) [inline Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/ContextMenu.js:28]: these `title` and `alt` were taken from Firefox

File: Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/ContextMenu.js [hunk 2]
```diff
@@ -13,20 +13,25 @@ window.__firefox__.includeOnce("ContextMenu", function() {
 
     var targetLink = target.closest("a");
     var targetImage = target.closest("img");
+                          
+    var data = {};
 
     if (!targetLink && !targetImage) {
+      // No link or image was tapped, sending empty callback so the ContextMenu will know that it
+      // should not show the menu(only text selection menu if applicable).
+      webkit.messageHandlers.contextMenuMessageHandler.postMessage(data);
```
Comment (iccub) [inline Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/ContextMenu.js:22]: This added by me, comment explains why

File: Client/Frontend/Browser/Tab.swift [hunk 3]
```diff
@@ -503,10 +503,6 @@ class Tab: NSObject {
         guard let url = self.webView?.url else {
             return
         }
-        
-        if let helper = contentScriptManager.getContentScript(ContextMenuHelper.name()) as? ContextMenuHelper {
-            helper.replaceWebViewLongPress()
```
Comment (iccub) [inline Client/Frontend/Browser/Tab.swift:508]: We don't need it anymore, it's added in BVC

File: Client/Frontend/Browser/ContextMenuHelper.swift [hunk 4]
```diff
@@ -126,6 +125,9 @@ extension ContextMenuHelper: TabContentScript {
         guard let data = message.body as? [String: AnyObject] else {
             return
         }
+        
+        // No image or link was pressed, we are clearing `elements` so no accidental link tap occurs.
+        if data.isEmpty { elements = nil }
```
Comment (iccub) [inline Client/Frontend/Browser/ContextMenuHelper.swift:130]: the context menu javascript trigger on regular taps too, so if you swipe and touch an image/link by accident, `elements` was assigned and caused a bug when context menu was showing even if you didn't tap on a link

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**